### PR TITLE
Support features in webxr

### LINKED ISSUE #24196: Support input profiles in WebXR
https://immersive-web.github.io/webxr/#dom-xrinputsource-profiles

Work on profiles is still ongoing at https://github.com/immersive-web/webxr-input-profiles , but once that's settled we should have our input sources report the correct profile strings.

### PR DESCRIPTION
**Pull request by Manishearth**
Based on https://github.com/servo/webxr/pull/119

Todo: 
 - [x] gate reference space creation on feature presence
 - [x] Fix the `features_deviceSupport` test to correctly use simulateUserActivation

Fixes #24196, #24270
r? @jdm @asajeffrey

### CODE REVIEW

**Comment by Manishearth** [comment]
Ready for review.

**Comment by jdm** [comment]
```
{"path": "webxr/xrSession_features_deviceSupport.https.html", "message": "Whitespace at EOL", "lineno": 34, "rule": "TRAILING WHITESPACE"}
{"path": "webxr/xrSession_features_deviceSupport.https.html", "message": "Whitespace at EOL", "lineno": 52, "rule": "TRAILING WHITESPACE"}
{"path": "webxr/resources/webxr_util.js", "message": "Whitespace at EOL", "lineno": 105, "rule": "TRAILING WHITESPACE"}
```

File: components/script/dom/fakexrdevice.rs
Comment (asajeffrey) [inline components/script/dom/fakexrdevice.rs]: Can you delete `profiles` now?

**Review by asajeffrey — APPROVED** [review]
LGTM once we've landed the webxr PR.

**Pull request merged**
